internal/provider/provider.go

```diff
@@ -318,6 +320,74 @@ func New() *schema.Provider {
 					},
 				},
 			},
+			"coder_metadata": {
+				Description: "Use this resource to attach key/value pairs to a resource. They will be " +
+					"displayed in the Coder dashboard.",
+				CreateContext: func(c context.Context, resourceData *schema.ResourceData, i interface{}) diag.Diagnostics {
+					resourceData.SetId(uuid.NewString())
+
+					pairs, err := populateIsNull(resourceData)
+					if err != nil {
+						return errorAsDiagnostics(err)
+					}
+					err = resourceData.Set("pair", pairs)
+					if err != nil {
+						return errorAsDiagnostics(err)
+					}
+
+					return nil
+				},
+				ReadContext: func(c context.Context, resourceData *schema.ResourceData, i interface{}) diag.Diagnostics {
+					return nil
+				},
+				DeleteContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
+					return nil
+				},
+				Schema: map[string]*schema.Schema{
+					"resource_id": {
+						Type:        schema.TypeString,
+						Description: "The \"id\" property of another resource that metadata should be attached to.",
+						ForceNew:    true,
+						Required:    true,
+					},
+					"pair": {
+						Type:     schema.TypeList,
+						ForceNew: true,
+						Required: true,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"key": {
+									Type:        schema.TypeString,
+									Description: "The key of this metadata item.",
+									ForceNew:    true,
+									Required:    true,
+								},
+								"value": {
+									Type:        schema.TypeString,
+									Description: "The value of this metadata item.",
+									ForceNew:    true,
+									Optional:    true,
+								},
+								"sensitive": {
+									Type: schema.TypeBool,
+									Description: "Set to \"true\" to for items such as API keys whose values should be " +
+										"hidden from view by default. Note that this does not prevent metadata from " +
+										"being retrieved using the API, so it is not suitable for secrets that should " +
+										"not be exposed to workspace users.",
+									ForceNew: true,
+									Optional: true,
+									Default:  false,
+								},
+								"is_null": {
```

Comment:
Would we just assume an empty string is null? Maybe that's too lossy?

Reply:
Possibly! I implemented it this way because the original ticket asked for null support, and I figured somebody who's used to Terraform's semantics wouldn't necessarily be expecting us to conflate null and "" the way Go does. But if we're OK with that behavior, I definitely wouldn't mind getting rid of this hack.